generator/codegen/collection_extensions.go

```diff
@@ -207,12 +208,13 @@ func getCollectionSchema(swagger *openapi3.T, appJSON *openapi3.MediaType, endpo
 	return
 }
 
-func newCollection(collectionRef, itemRef string) Collection {
+func newCollection(collectionRef, itemRef string, noPagination bool) Collection {
```

Comment:
Instead of using flags would it be possible to have two constructors? So add

```go
func newCollectionWithoutPagination(collectionRef, itemRef string) Collection 
```

Then do the check when you choose which constructor to call?

I think having Boolean flags can make the code harder to read as it isn't clear what the code is doing without diving into it if you just see `true` or `false` in a random function call.

The other option would be to add some `opts` and and have them as optional arguments to the function, but that might be overkill in this case https://github.com/uber-go/guide/blob/master/style.md#functional-options